views/reducers/Camera.js

```diff
@@ -0,0 +1,48 @@
+import ACTIONS from '../actions';
+
+const Three = require('three');
+
+const InitialState = {
+  config: {
+    fov: 75,
+    aspect: window.innerWidth / window.innerHeight,
+    near: 1,
+    far: 800,
+    position: new Three.Vector3(400, 0, 400),
+    lookat: new Three.Vector3(0, 0, 0),
+  },
+};
+
+const mapRotationVectorFromAction = (state, action) => {
+  // TODO: Add factory for Three Objects
+  const speed = action.speed || 1;
+  switch (action.rotation) {
+    case ACTIONS.MOVE.UP:
+      return state.config.lookat.add(new Three.Vector3(0, 1 * speed, 0));
```

Comment:
this is mutating the previous state!!!  Should be new Vector3(...).add(state.config.lookat),